src/imitation/rewards/discrim_net.py

```diff
@@ -33,32 +33,20 @@ def __init__(self):
     # TODO(adam): this is only used by AIRL; sad we have to always include it
     self._log_policy_act_prob_ph = tf.placeholder(
         shape=(None,), dtype=tf.float32, name="log_ro_act_prob_ph")
-    self.build_graph()
-
-    # check that required attributes have been set
-    required_attrs = [
-      '_disc_loss', '_policy_train_reward', '_policy_test_reward',
-    ]
-    for req_attr in required_attrs:
-      assert hasattr(self, req_attr), \
-        f"required attr .{req_attr} not added by {type(self)}" \
-        ".build_graph()"
 
-  @property
-  def disc_loss(self):
-    return self._disc_loss  # pytype: disable=attribute-error
+    self._disc_loss = None  # type: tf.Tensor
+    self._policy_train_reward = None  # type: tf.Tensor
+    self._policy_test_reward = None  # type: tf.Tensor
+    self._disc_logits = None  # type: tf.Tensor
 
-  @property
-  def policy_train_reward(self):
-    return self._policy_train_reward  # pytype: disable=attribute-error
+    self.build_graph()
 
-  @property
-  def policy_test_reward(self):
-    return self._policy_test_reward  # pytype: disable=attribute-error
+    assert self._disc_loss is not ...
```

Comment:
Why ellipsis and not `None`? Is this a `typing` convention?

Reply:
This is an error from when I was messing around with trying to get PyCharm inspection to be happy with my types

Reply:
Hmmm looks like I had already fixed a day before you made this comment.
You seem to have missed https://github.com/HumanCompatibleAI/imitation/pull/156/commits/a6e6f55e0fa10b207ddf45b145e12fdf518b1127 where I also made some minor log name changes that you might want to see.